src/cascadia/WinRTUtils/inc/WtExeUtils.h

```diff
@@ -104,3 +104,29 @@ _TIL_INLINEPREFIX const std::wstring& GetWtExePath()
     }();
     return exePath;
 }
+
+// Method Description:
+// - Escapes the given string so that it can be used as a command-line arg within a quoted string.
+// - e.g. given `";foo\` will return `\"\;foo\\` so that the caller can construct a command-line
+//   using something such as `fmt::format(L"wt --title \"{}\"", EscapeCommandlineArg(TabTitle()))`.
+// Arguments:
+// - arg - the command-line argument to escape.
+// Return Value:
+// - the escaped command-line argument.
+_TIL_INLINEPREFIX std::wstring EscapeCommandlineArg(const std::wstring_view arg)
+{
+    std::wstringstream stream;
+    for (auto it = arg.cbegin(); it != arg.cend(); ++it)
+    {
+        if (*it == L';' || *it == L'"')
+        {
+            stream << L'\\';
+        }
+        stream << *it;
+    }
+    if (arg.back() == L'\\')
+    {
+        stream << L'\\';
+    }

```

Comment:
This function doesn't handle escaping of backslashes correctly. This should be correct:
```cpp
_TIL_INLINEPREFIX std::wstring EscapeCommandlineArg(const std::wstring_view& arg)
{
    std::wstring out;
    out.reserve(arg.size() + 2);
    out.push_back(L'"');

    size_t backslashes = 0;
    for (const auto ch : arg) {
        if (ch == L'\\') {
            backslashes++;
        } else {
            if (ch == L';' || ch == L'"') {
                out.append(backslashes + 1, L'\\');
            }
            backslashes = 0;
        }
        out.push_back(x);
    }

    out.append(backslashes, L'\\');
    out.push_back(L'"');
    return out;
}
```

I've included the addition of quotes to the function, because I felt this to be more consistent.
If you dislike that please remove the two occurrences of `out.push_back(L'"');` and the `+ 2`.

Reply:
Do the back-slashes need escaping? Opening a folder `D:\Downloads\;SemicolonStart` works fine as-is.

![image](https://user-images.githubusercontent.com/5821575/134635371-3dc07f73-121e-4fd3-a8f1-316ee65e3c8c.png)

![image](https://user-images.githubusercontent.com/5821575/134635399-55473574-1f69-4d61-beae-5080bf99942c.png)

Side-note: I didn't realise C++ had support for for-each loops like in Java. I see that came in C++11.

Reply:
Ah it's quotes that are the problem, not semicolons. Will add a test and fix.
```json
{
    "command": 
    {
        "action": "newWindow",
        "tabTitle": "\\\";foo\\"
    },
    "keys": "ctrl+shift+s"
}
```

![image](https://user-images.githubusercontent.com/5821575/134638142-d5e7306e-fdc6-4e73-b4a5-0f7f5787b09f.png)

Reply:
Done

Reply:
> Do the back-slashes need escaping?

Yeah the command lines rules on Windows are just outright weird, but the gist is that `a\\\b` literally contains 3 backslashes. This is utterly confusing of course, but it's the way it is...
This means you need to escape slashes _if_ the following character is a character that needs escaping.
So `\\\b` needs to be escaped as `\\\b` (no escaping), whereas `\\\"` needs to be escaped as `\\\\\\\"` (escaping for all backslashes + the quote).